### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by rhi-batstone**
Script to transform life expectancy data into the format required for CA Profiles

### CODE REVIEW

File: source/life_expectancy_script.R [hunk 1]
```diff
@@ -0,0 +1,142 @@
+library(tidyverse)
+library(readxl)
+
+file_path <- "data/life-expectancy/life-expectancy-17-19-tab3.xlsx"
```
Comment (jsphdms) [inline source/life_expectancy_script.R:4]: I'd maybe move the spreadsheet up one level so it sits next to the main spreadsheet

File: source/life_expectancy_script.R [hunk 2]
```diff
@@ -0,0 +1,142 @@
+library(tidyverse)
+library(readxl)
```
Comment (jsphdms) [inline source/life_expectancy_script.R:2]: Would it be better to move all library calls to the main script?

File: source/life_expectancy_script.R [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    "85 to 89",
+    "90 and over"
+  )
+
+final_clean <- list() # Empty list to add the new dataframes into
+
+for (i in seq_along(clean_sheets)) {
+  #TODO maybe this could be a list instead of 8 new objects
+  
+  # The tables are in multiples of 23 so every 23 columns is a new table (each tables contains 4 council areas in alphabetical order)
```
Comment (jsphdms) [inline source/life_expectancy_script.R:68]: Could you check if you've linted? This line looks over 80 characters

File: source/life_expectancy_script.R [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+sheets <-
+  file_path %>%
+  excel_sheets() %>%
+  purrr::set_names() %>%
+  purrr::map(read_excel, path = file_path)
+
+# Transforming the sheets --------------------------------------------------------
+
+sheets <- sheets[-c(1, 2)] # removing the metadata sheets
```
Comment (jsphdms) [inline source/life_expectancy_script.R:16]: Could this line be incorporated into the pipeline above? E.g. `%>% `[`(-c(1, 2))`

File: source/life_expectancy_script.R [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+transposed_sheets <- lapply(sheets, function(x) {
```
Comment (jsphdms) [inline source/life_expectancy_script.R:18]: This is an optional nit-pick: for naming I prefer suffixes over prefixes because that helps with autocomplete.

File: source/life_expectancy_script.R [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+})
+
+for (i in seq_along(transposed_sheets)) {
+  row.names(transposed_sheets[[i]]) <-
+    NULL # Remove row names left over from transpose
+}
+
+clean_sheets <- lapply(transposed_sheets, function(x)
+  x %>%
+    select(1:161) %>% # 161 is where the data stops and it becomes footnotes
```
Comment (jsphdms) [inline source/life_expectancy_script.R:32]: Very handy comment!

File: source/life_expectancy_script.R [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+    gather("0":"90 and over", key = "age", value = "life_expectancy") %>% # Combine age columns into one
+    distinct() # Remove duplicate rows (West Lothian, being at the end of the list, has dupes from zoo::na.locf() )
+  
+  final_clean[i] <-
+    list(all_combined) # Create a new list of all the cleaned data for each year range
+}
+
+life_expectancy_list = list()
+
+for (i in seq_along(final_clean)) {
```
Comment (jsphdms) [inline source/life_expectancy_script.R:110]: Is a loop needed for this?

**Review by jsphdms — CHANGES_REQUESTED** [review]
Thanks! This is a good start to a rather complicated data cleaning job - it's commented well and makes good use of existing code. I think this might need a second review after this one (mainly because I've not had a chance to run it myself).

A few general comments:
- At the moment there are quite a few intermediate variables which makes it a bit hard to follow. Could these be collapsed into a smaller number?
- Could you give the PR a slightly more descriptive name?
- Does this script need to be sourced in the main script?

File: source/life_expectancy_script.R (hunk 7 above)
Comment (rhi-batstone) [reply source/life_expectancy_script.R:110]: I think so? I need to create the 7 subsets for each sheet. Can't think of another way to do it.

File: source/life_expectancy_script.R (hunk 2 above)
Comment (rhi-batstone) [reply source/life_expectancy_script.R:2]: Do you want this life_expectancy_script.R to be sourced in the 0-Parameterised_reports.R script?

Comment (jsphdms) [reply source/life_expectancy_script.R:2]: Yes I think so. We're aiming to keep it as simple as possible for whoever runs this in future.
